File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/TVA.Phasors/ChannelParsingStateBase.vb
'*******************************************************************************************************
'  ChannelParsingStateBase.vb - Parsing state base class
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  3/7/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

''' <summary>This class represents the common implementation of the protocol independent parsing state class used by any kind of data.</summary>
''' <remarks>This class is inherited by subsequent classes to provide parsing state information particular to a data type's needs.</remarks>
Public MustInherit Class ChannelParsingStateBase

    Implements IChannelParsingState

    ' This is expected to be overriden by the final derived class
    Public MustOverride ReadOnly Property DerivedType() As Type Implements IChannelParsingState.DerivedType

    Public Overridable ReadOnly Property This() As IChannelParsingState Implements IChannelParsingState.This
        Get
            Return Me
        End Get
    End Property

End Class

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/ChannelParsingStateBase.cs
//*******************************************************************************************************
//  ChannelParsingStateBase.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  3/7/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using System;

namespace PCS.PhasorProtocols
{
    /// <summary>
    /// Represents the common implementation of the protocol independent parsing state class used by any kind of data.<br/>
    /// This is the base class of all parsing state classes in the phasor protocols library;
    /// it is the root of the parsing state class hierarchy.
    /// </summary>
    /// <remarks>
    /// This class is inherited by subsequent classes to provide parsing state information particular to data type needs.
    /// </remarks>
    public abstract class ChannelParsingStateBase : IChannelParsingState
    {
        #region [ Members ]

        // Fields
        private int m_parsedBinaryLength;

        #endregion

        #region [ Properties ]

        /// <summary>
        /// Gets or sets the length of the associated <see cref="IChannel"/> object being parsed from the binary image.
        /// </summary>
        public virtual int ParsedBinaryLength
        {
            get
            {
                return m_parsedBinaryLength;
            }
            set
            {
                m_parsedBinaryLength = value;
            }
        }

        #endregion
    }
}